### ISSUE TITLE
实现ResNet50模型训练代码（基于TensorFlow）

### ISSUE THREAD

**Issue by liuzhenzzz** [liuzhenzzz/zz_repository#1]
## 需求描述
请实现一个基于TensorFlow框架的ResNet50模型训练代码，需满足以下要求：

1. 能对指定的图像分类数据集进行训练和验证。
2. 具备学习率优化功能（如学习率衰减/调度器等）。
3. 具备较高的精度（推荐使用迁移学习与数据增强提升效果）。
4. 代码需有防止过拟合的措施（如正则化、数据增强、EarlyStopping等）。
5. 数据集路径如下：
   - 训练集路径：`./classification_datasets/train`
   - 验证集路径：`./classification_datasets/val`
6. 代码整体结构清晰，便于后续维护和扩展。

## 建议实现思路
- 使用`tf.keras.applications.ResNet50`作为主模型，采用ImageNet预训练权重。
- 使用`ImageDataGenerator`进行数据增强。
- 使用`ReduceLROnPlateau`或`LearningRateScheduler`进行学习率优化。
- 使用`EarlyStopping`、`Dropout`和`L2正则化`等方法防止过拟合。
- 训练/验证集分别用`flow_from_directory`加载。
- 代码需包含完整的训练、验证和保存模型流程。

## 参考代码结构
```python
import tensorflow as tf
from tensorflow.keras.preprocessing.image import ImageDataGenerator
from tensorflow.keras.applications import ResNet50
from tensorflow.keras.layers import Dense, Dropout, GlobalAveragePooling2D
from tensorflow.keras.models import Model
from tensorflow.keras.optimizers import Adam
from tensorflow.keras.callbacks import EarlyStopping, ReduceLROnPlateau

train_dir = './classification_datasets/train'
val_dir = './classification_datasets/val'

# 数据增强
train_datagen = ImageDataGenerator(
    rescale=1./255,
    rotation_range=20,
    width_shift_range=0.2,
    height_shift_range=0.2,
    shear_range=0.2,
    zoom_range=0.2,
    horizontal_flip=True,
    fill_mode='nearest')
val_datagen = ImageDataGenerator(rescale=1./255)

train_generator = train_datagen.flow_from_directory(
    train_dir,
    target_size=(224, 224),
    batch_size=32,
    class_mode='categorical')
val_generator = val_datagen.flow_from_directory(
    val_dir,
    target_size=(224, 224),
    batch_size=32,
    class_mode='categorical')

base_model = ResNet50(weights='imagenet', include_top=False, input_shape=(224,224,3))
for layer in base_model.layers:
    layer.trainable = False

x = base_model.output
x = GlobalAveragePooling2D()(x)
x = Dropout(0.5)(x)
x = Dense(256, activation='relu', kernel_regularizer=tf.keras.regularizers.l2(0.01))(x)
x = Dropout(0.5)(x)
predictions = Dense(train_generator.num_classes, activation='softmax')(x)
model = Model(inputs=base_model.input, outputs=predictions)

model.compile(optimizer=Adam(learning_rate=1e-4),
              loss='categorical_crossentropy',
              metrics=['accuracy'])

callbacks = [
    EarlyStopping(monitor='val_loss', patience=10, restore_best_weights=True),
    ReduceLROnPlateau(monitor='val_loss', factor=0.2, patience=5, min_lr=1e-6)
]

model.fit(
    train_generator,
    epochs=50,
    validation_data=val_generator,
    callbacks=callbacks)

model.save('resnet50_tf_best.h5')
```
请根据实际需求进行细节调整和完善。
